### PR TITLE
:lipstick: :sparkles: Created menu solutions component on the header #186

### LINKED ISSUE #186: 🎨 Create Menu Solutions Component on the Header.
#  Create the menu solutions component
---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user, I want a menu on the header section that opens when the solutions tab is clicked and drops from the top (if possible animate this) that captures my attention and introduces the menu content with router links to the different pages shown in the design below. I want a good experience on the web and phone. The menu section is a component created in `libs\elements\layout\header` inside the components folder. If you need to update the header component to achieve this feel free to do so.

Design:

menu xcali

Link to full design:

```
https://xd.adobe.com/view/b98caf27-6549-49e5-8ed8-c5234788df61-ecdf/screen/cc35e19a-24f2-42ce-8803-ee2942839528/
```
 
### Technical Analysis

This feature can be found in `libs\elements\layout\header` inside a components folder. The component's name is header-menu. The individual router links that are visible when the menu is open should be a reusable menu-item component located in the same module in the components folder.

menu-item

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the design above.
2. The component is clean and does not have more than 30 lines
3. The component SCSS uses CSS variables for colour, font, ... Connected to the global style.scss
5. The design is responsive for screen, tablet and phone

### PR DESCRIPTION
**Pull request by levy-web**
# Description

created menu item with links to other pages that is displayed when solution nav item is clicked. It toggles displaying of the menu items.

Reference the issue related to this PR as shown below. Every issue has it's own number you can find the issue numbers [here](https://github.com/italanta/italanta-apps/issues).

Fixes #186

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## Screenshot (optional)

Screenshot from 2023-09-07 22-20-51

Screenshot from 2023-09-11 18-29-03

Screenshot from 2023-09-11 18-28-33

[Screencast from 07-09-2023 10:21:52 WB.webm](https://github.com/italanta/elewa-website/assets/71401172/ca77ad5a-631d-487f-893c-b615ec4e4af4)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**: (optional)

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/elements/layout/header/src/lib/components/header-menu/header-menu.component.scss [hunk 1]
```diff
@@ -0,0 +1,25 @@
+.menu-items{
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    max-width: 1200px;
+    margin-inline: auto;
+    z-index: 50000000000000;
+    background-color: white;
```
Comment (AnnaL001) [inline libs/elements/layout/header/src/lib/components/header-menu/header-menu.component.scss:8]: For reuse of colors, use CSS variables specified in elements/theming/_variables.scss  for specifying colors and background colors

File: libs/elements/layout/header/src/lib/components/header-menu/header-menu.component.scss [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        a {
+            color: inherit;
+            text-decoration: none;
+        }
+    }
+
+}
+
+.active{
+    background-color: #eaeaea;
```
Comment (AnnaL001) [inline libs/elements/layout/header/src/lib/components/header-menu/header-menu.component.scss:23]: Same here

File: libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss [hunk 3]
```diff
@@ -0,0 +1,96 @@
+.booklets-design{
+    width: 24px;
+    height: 36px;
+    margin-top: 5px;
+    border-radius: 110px;
+    transform: matrix(0, -1, 1, 0, 0, 0);
+    background: transparent linear-gradient(307deg, #F2AD69 0%, #F2AD6900 100%) 0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss:7]: Same here for the linear gradient

File: libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+.conversational-Learning-design{
+
+    display: flex;
+
+    .conversational-Learning-one{
+        width: 24px;
+        height: 36px;
+        margin-top: 5px;
+        transform: matrix(0, -1, 1, 0, 0, 0);
+        background: transparent linear-gradient(180deg, #E69083 0%, #E6908300 100%) 0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss:19]: Same here

File: libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        border-radius: 110px;
+        opacity: 1;
+    }
+    .conversational-Learning-two{
+        width: 24px;
+        height: 36px;
+        transform: matrix(0, 1, -1, 0, 0, 0);
+        background: transparent linear-gradient(180deg, #E69083 0%, #E6908300 100%) 0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss:27]: Same here

File: libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    display: flex;
+    gap: 5px;
+    margin-top: 5px;
+
+    .content-development-one{
+        width: 17px;
+        margin-top: auto;
+        height: 16px;
+        background: transparent linear-gradient(180deg, #F2CF73 0%, #F2CF7300 100%) 0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss:44]: Same here

File: libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        mix-blend-mode: multiply;
+        border-radius: 110px;
+    }
+
+    .content-development-two{
+        width: 17px;
+        margin-top: auto;
+        height: 25px;
+        background: transparent linear-gradient(180deg, #F2CF73 0%, #F2CF7300 100%) 0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss:53]: Same here

File: libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        mix-blend-mode: multiply;
+        border-radius: 110px;
+        opacity: 1;
+    }
+
+    .content-development-three{
+        width: 16px;
+        height: 33px;
+        margin-top: auto;
+        background: transparent linear-gradient(180deg, #F2CF73 0%, #F2CF7300 100%) 0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss:63]: Same here

File: libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+.cosultancy-design{
+    display: flex;
+    height: 33px;
+    margin-top: 5px;
+
+    .cosultancy-one{
+        position: absolute;
+        width: 32px;
+        height: 32px;
+        background: transparent linear-gradient(90deg, #DBC2F500 0%, #DBC2F5E6 100%) 0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss:79]: Same here

File: libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+        border-radius: 50%;
+
+    }
+    .cosultancy-two{
+        position: absolute;
+        width: 32px;
+        margin-left: 20px;
+        height: 32px;
+        transform: matrix(-1, 0, 0, -1, 0, 0);
+        background: transparent linear-gradient(90deg, #DBC2F500 0%, #DBC2F5E6 100%) 0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/elements/layout/header/src/lib/components/menu-item/menu-item.component.scss:91]: Same here

**Review by AnnaL001 — CHANGES_REQUESTED** [review]
Nicely done, just a few changes

**Comment by levy-web** [comment]
@AnnaL001 done with the fixes

**Review by LemmyMwaura — APPROVED** [review]

**Pull request merged**